tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousCSTest.java

```diff
@@ -144,205 +112,57 @@ public void testClassLevelAsyncIsNotFinished() {
-     * Using this factory method ensures they will be completed at the end of
-     * the test if your test fails.
-     */
-    private CompletableFuture<Void> newWaitingFuture() {
-        CompletableFuture<Void> result = new CompletableFuture<>();
-        waitingFutures.add(result);
-        return result;
-    }
-
-    /**
-     * A helper method to complete a waiting future with a more readable syntax
-     */
-    private void complete(CompletableFuture<?> future) {
-        future.complete(null);
-    }
-
-    private void handleCompletionException(CompletionException e) throws AssertionError {
-        throw new AssertionError("testAsync: unexpected Exception calling service: "
-                + e.getCause().getMessage(), e);
-    }
-    
-    private static Connection waitUntilCompleted(CompletionStage<Connection> resultFuture) {
-        return CompletableFutureHelper.toCompletableFuture(resultFuture).join();
-    }
-    
-    /**
-     * Tests whether the given CompletionStage completes within 500ms
-     * <p>
-     * Used to avoid a race condition where a test wants to take some action which
-     * should cause the CompletionStage to complete, but asynchronous execution
-     * means it may not happen immediately.
-     */
-    private static boolean completesQuickly(CompletionStage<Connection> resultFuture) {
-        try {
-            CompletableFutureHelper.toCompletableFuture(resultFuture).get(500, TimeUnit.MILLISECONDS);
-        }
-        catch (TimeoutException ex) {
-            // Did not complete quickly
-            return false;
-        }
-        catch (Exception ex) {
-            // Completed with exception, fall through, we don't care about the result
-        }
-        return true;
-    }
-
-    private static boolean isCancelled(CompletionStage<Connection> resultFuture) {
-        return CompletableFutureHelper.toCompletableFuture(resultFuture).isCancelled();
-    }
-
-    private static boolean isCompletedExceptionally(CompletionStage<Connection> resultFuture) {
-        return CompletableFutureHelper.toCompletableFuture(resultFuture).isCompletedExceptionally();
-    }
-    
-    private static void assertThrowsExecutionExceptionWithCause(Class<? extends Throwable> causeClazz, ThrowingRunnable runnable) {
+        Task taskToPerform = new ServiceTask();
+        CompletionStage<Task> taskResult = client.serviceCS(taskToPerform);
         try {
-            runnable.run();
-            Assert.fail("ExecutionException not thrown");
-        }
-        catch (ExecutionException ex) {
-            Assert.assertTrue(causeClazz.isInstance(ex.getCause()), "Cause of ExecutionException was " + ex.getCause());
-        }
-        catch (Throwable ex) {
-            Assert.fail("Unexpected exception thrown", ex);
-        }
-    }
+            with().pollThread(Thread::new).await()
+                .atMost(120, TimeUnit.MILLISECONDS).until(
+                () -> taskToPerform.getTaskResult() == "service DATA");
```

Comment:
Use `.equals()` rather than `==` to compare strings

`"service DATA".equals(taskToPerform.getTaskResult())` is probably the safest option here to avoid an exception if `getTaskResult()` returns `null`.